src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -49,15 +45,14 @@ template <typename DeviceType>
 struct DistributedSearchTreeImpl
 {
   // spatial queries
-  template <typename ExecutionSpace, typename Predicates>
-  static void
-  queryDispatch(SpatialPredicateTag,
-                DistributedSearchTree<typename DeviceType::memory_space,
-                                      void> const &tree,
+  template <typename DistributedTree, typename ExecutionSpace,
+            typename Predicates, typename Indices, typename Offset,
+            typename Ranks>
+  static std::enable_if_t<Kokkos::is_view<Indices>{} &&
+                          Kokkos::is_view<Offset>{} && Kokkos::is_view<Ranks>{}>
+  queryDispatch(SpatialPredicateTag, DistributedTree const &tree,
                 ExecutionSpace const &space, Predicates const &queries,
-                Kokkos::View<int *, DeviceType> &indices,
-                Kokkos::View<int *, DeviceType> &offset,
-                Kokkos::View<int *, DeviceType> &ranks)
```

Comment:
We should probably also check that these views have the correct rank (here and below).

Reply:
Yes